docs/src/architecture.md

```diff
@@ -1,30 +1,166 @@
 # Architecture
 
-For High Availability and Scalability goals, the PostgreSQL database management
-system provides administrators with built-in **physical replication**
-capabilities based on **Write Ahead Log (WAL) shipping**.
+This section covers the main architectural aspects you need to consider
+when deploying PostgreSQL in Kubernetes.
+
+!!! Important
+    If you are deploying PostgreSQL in a self-managed Kubernetes environment,
+    please make sure you read the ["Kubernetes architecture"](#kubernetes-architecture)
+    below when you start planning your journey to the Cloud Native world.
+
+## Synchronizing the state
+
+PostgreSQL is a database management system and, as such, it needs to be treated
+as a **stateful workload** in Kubernetes. While, for High Availability (HA) and
+Disaster Recovery (DR) purposes, stateless applications primarily take care of
+traffic redirection, in the case of a database its state must be replicated in
+multiple locations, preferably in a continuous and instantaneous way, by adopting
+either of the following two strategies:
+
+* *storage-level replication*, normally persistent volumes
+* *application-level replication*, in this specific case PostgreSQL
```

Comment:
here I want to check with you what we mean to express.
As it stands, seems to me a tiny overview of replication in Kubernetes for stateful vs stateless.

Reply:
The point here is to clarify that normally state is replicated via storage. For PostgreSQL we should avoid volume replication, and recommend application level replication. I have noticed that from an architecture's point of view, we take it for granted, but then, decisions about storage have already been taken and cannot be changed. The goal here is to make users aware of that - and it is highlighted later in the doc when we mention shared nothing architectures.